### PR TITLE
GROUP MANAGEMENT: Applies logic to render wizard vs Create Group button

### PR DESCRIPTION
**Pull request by njons**
#### Description:
Adds logic to render create group wizard (to help guide the creation of a first group) and the "create group" button for subsequent creation of groups.  

**Summary**: 

1.  Work on existing `<Groups/>`: 
    - Mock of upcoming component `<DataPanel/>` added as function (`RenderDataPanel`)
    - `this.state.wizardCreateGroup` renamed to `this.state.firstGroup`
    - Method `renderCreateButton` added to sets `state.firstGroup: false`

2. New component `<WizardPanel/>`:
    - Handles rendering of the correct wizard at the right time:
        - Mock of **create first group wizard** added as function (`RenderFirstGroupWizard`) - mock inline styling also applied for clarity, but many elements not functional
        - In preparation for the **reply to first invite wizard**, the state `this.state = { firstInvite: true };` is ready to handle rendering of appropriate invites wizard (first or not)
3. Implemented logic between `<Groups/>` (parent) and `<WizardPanel/>` (child) :
- I have kept this logic local as I don't think the entire app needs to know that a user has completed a wizard and that is why I have avoided the redux store: 
        1.  CREATE GROUP button (`<WizardPanel/>`)
        2.  `renderCreateButton()` method (`<Groups/>`) 
        3. `this.state.firstGroup: false` (`<Groups/>`) 
        4. `this.props.firstGroup` is passed back to child (`<WizardPanel/>`) 

---
###### Groups (before first group is created = wizard)
Screenshot 2020-09-01 at 16 15 11

###### Groups (after first group is created = )
Screenshot 2020-09-01 at 16 14 58

---

#### For reviewer:
**WHAT:**
- [x] Check that wizard disappears when "CREATE GROUP" is clicked (there is also console.logs) 
- [x] Check that create group button appears next to "Groups" heading

**HOW:**
- [ ] Read the above description
- [ ] Reviewed the code changes
- [ ] Navigate to the branch (pulled the latest version)
- [ ] Run the code and been able to execute the expected function
- [ ] Check any styling on both desktop and mobile sizes

### CODE REVIEW

File: src/login/components/App/DashboardApp/Settings/Groups/Groups.js [hunk 1]
```diff
@@ -33,7 +34,25 @@ class Groups extends Component {
   constructor(props) {
     super(props);
     // will start as true and permanently be set to false when wizard is completed
-    this.state = { wizardCreateGroup: true};
+    this.state = { firstGroup: true };
+    this.renderCreateButton = this.renderCreateButton.bind(this);
```
Comment (eunjuhuss) [inline src/login/components/App/DashboardApp/Settings/Groups/Groups.js:47]: please, remove` this.renderCreateButton = this.renderCreateButton.bind(this);`

File: src/login/components/App/DashboardApp/Settings/Groups/Groups.js [hunk 2]
```diff
@@ -73,7 +73,10 @@ class Groups extends Component {
               third-party services using eduID for login.
             </p>
             <RenderDataPanel />
-            <WizardPanel />
+            <WizardPanel
+              renderCreateButton={this.renderCreateButton}
```
Comment (eunjuhuss) [inline src/login/components/App/DashboardApp/Settings/Groups/Groups.js:91]: could we change to arrow function `renderCreateButton={this.renderCreateButton}` ->` renderCreateButton={()=>this.renderCreateButton()}`

File: src/login/components/App/DashboardApp/Settings/Groups/Groups.js [hunk 3]
```diff
@@ -33,7 +34,25 @@ class Groups extends Component {
   constructor(props) {
     super(props);
     // will start as true and permanently be set to false when wizard is completed
-    this.state = { wizardCreateGroup: true};
+    this.state = { firstGroup: true };
+    this.renderCreateButton = this.renderCreateButton.bind(this);
+  }
+
+  renderCreateButton() {
```
Comment (eunjuhuss) [inline src/login/components/App/DashboardApp/Settings/Groups/Groups.js:50]: could we change to arrow function ` renderCreateButton()`  -> `renderCreateButton=()=>`

**Review by eunjuhuss — COMMENTED** [review]
I commented minor comments after we discussed about arrow function.

File: src/login/components/App/DashboardApp/Settings/Groups/Groups.js (hunk 1 above)
Comment (njons) [reply src/login/components/App/DashboardApp/Settings/Groups/Groups.js:47]: 👍

File: src/login/components/App/DashboardApp/Settings/Groups/Groups.js (hunk 2 above)
Comment (njons) [reply src/login/components/App/DashboardApp/Settings/Groups/Groups.js:91]: 👍

File: src/login/components/App/DashboardApp/Settings/Groups/Groups.js (hunk 3 above)
Comment (njons) [reply src/login/components/App/DashboardApp/Settings/Groups/Groups.js:50]: 👍

File: src/login/components/App/DashboardApp/Settings/Groups/Groups.js (hunk 2 above)
Comment (eunjuhuss) [reply src/login/components/App/DashboardApp/Settings/Groups/Groups.js:91]: I see the code sending function to wizardPanel but I wonder how "renderCreateButton" is working.

**Pull request merged**